diff --git a/node_modules/react-native-document-picker/android/src/main/java/io/github/elyx0/reactnativedocumentpicker/DocumentPickerModule.java b/node_modules/react-native-document-picker/android/src/main/java/io/github/elyx0/reactnativedocumentpicker/DocumentPickerModule.java
index 999153a..d7fb9eb 100644
--- a/node_modules/react-native-document-picker/android/src/main/java/io/github/elyx0/reactnativedocumentpicker/DocumentPickerModule.java
+++ b/node_modules/react-native-document-picker/android/src/main/java/io/github/elyx0/reactnativedocumentpicker/DocumentPickerModule.java
@@ -102,6 +102,7 @@ public class DocumentPickerModule extends ReactContextBaseJavaModule {
 		try {
 			Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
 			intent.addCategory(Intent.CATEGORY_OPENABLE);
+      intent.putExtra("android.content.extra.SHOW_ADVANCED", true);
 
 			intent.setType("*/*");
 			if (!args.isNull(OPTION_TYPE)) {
